### ISSUE TITLE
OpenDrive map "does not work" since release v2.20.0 or more recent

### ISSUE THREAD

**Issue by mschneider1711** [esmini/esmini#221]
Hi,
I am not familiar with the OpenDrive format. I am just using the maps I am given. I am encountering an error with a map I did not use recently. Therefore I just encountered this issue now. From release v2.20.0 my Scenario can not be played and the log.txt gives me only this message (apart from the "normal" stuff):

> Signal Type crosswalk doesn't exists for this country
Failed to find object 
Exception: Failed to find object: 

I am struggling to understand the problem because in your release notes for v2.20.0 are not any changes to the OpenDrive module mentioned. Maybe you could provide quick help if possible.

**Comment by eknabevcc** [comment]
Would it be possible to get an example, or at least part of it? It's so much easier to investigate then.

**Comment by mschneider1711** [comment]
What works best for you, the scenario or open drive file? I have no idea where this error may have its root.

**Comment by mschneider1711** [comment]
Okay, I think I messed up there. The "Failed to find object" message should be from the AddEntityAction because it was not implemented "right" back in v2.20. I just reproduced the error with the newest release v2.20.3. And it gives me this:

> unknown lane road mark color: (null) (road id=0), set to standard (white)
unknown lane road mark color: (null) (road id=0), set to standard (white)
unknown lane road mark color: (null) (road id=0), set to standard (white)
unknown lane road mark color: (null) (road id=0), set to standard (white)

Probably problems with the open drive map then I think.

**Comment by mschneider1711** [comment]
So I checked again and would like to correct myself. So from release v2.20.2 I get the error message above.
The releases beforehand work fine.

> unknown lane road mark color: (null) (road id=0), set to standard (white)
unknown lane road mark color: (null) (road id=0), set to standard (white)
unknown lane road mark color: (null) (road id=0), set to standard (white)
unknown lane road mark color: (null) (road id=0), set to standard (white)

**Comment by eknabevcc** [comment]
Yes, it's the OpenDRIVE. Would be appreciated if you could send at least an excerpt. If you find this element <roadMark> please just paste the complete element content here. Example:
```xml
<roadMark color="standard" height="0.02" laneChange="both" sOffset="0" type="broken" weight="standard" width="0.2">
    <type name="broken" width="0.2">
        <line length="4.0" rule="caution" sOffset="0" space="8.0" tOffset="0" width="0.2"/>
    </type>
</roadMark>
```
And yes, roadmark color is one of recent changes, so there most likely a bug. Which we should fix asap.

**Comment by eknabevcc** [comment]
Yes, there was definitely a bug in esmini. The root cause was a missing `color` attribute in `roadMark` element.

The attribute is required (at least from OpenDRIVE v1.5) and normally present, why we haven't seen the bug until now. But of course esmini shouldn't crash, but instead warn and apply standard color if missing. It's fixed now by commit 40eb7b7fdf74dbdc98c81fc50e1ccfffc8d50a7e.

Thanks for helping us find the flaw! 

The fix will be included in next release. Do you need one asap?

**Comment by mschneider1711** [comment]
Very good, thank you very much as well. I need the release but not asap till it's not that important right now. It's totally fine if you take your time. Cheers.
